api/cas/util.py

```diff
@@ -0,0 +1,103 @@
+import json
+
+from rest_framework.exceptions import AuthenticationFailed
+
+import jwe
+import jwt
+
+from api.base import settings
+from api.base.authentication.drf import check_user
+from api.base.exceptions import (UnconfirmedAccountError, UnclaimedAccountError, DeactivatedAccountError,
+                                 MergedAccountError, InvalidAccountError)
+
+from addons.twofactor.models import UserSettings as TwoFactorUserSettings
+
+# user status
+USER_ACTIVE = 'USER_ACTIVE'
+USER_NOT_CLAIMED = 'USER_NOT_CLAIMED'
+USER_NOT_CONFIRMED = 'USER_NOT_CONFIRMED'
+USER_DISABLED = 'USER_DISABLED'
+USER_STATUS_INVALID = 'USER_STATUS_INVALID'
+
+# login and registration exceptions
+MISSING_CREDENTIALS = 'MISSING_CREDENTIALS'
+ACCOUNT_NOT_FOUND = 'ACCOUNT_NOT_FOUND'
+INVALID_PASSWORD = 'INVALID_PASSWORD'
+INVALID_VERIFICATION_KEY = 'INVALID_VERIFICATION_KEY'
+INVALID_ONE_TIME_PASSWORD = 'INVALID_ONE_TIME_PASSWORD'
+TWO_FACTOR_AUTHENTICATION_REQUIRED = 'TWO_FACTOR_AUTHENTICATION_REQUIRED'
+ALREADY_REGISTERED = 'ALREADY_REGISTERED'
+
+
+def decrypt_payload(body):
+    """
+    Decrypt the payload.
+
+    :param body: the JWE/JwT encrypted payload body
+    :return: the decrypted json payload
+    """
+
+    if not settings.API_CAS_ENCRYPTION:
+        try:
+            return json.loads(body)
+        except TypeError:
+            raise AuthenticationFailed
+    try:
+        payload = jwt.decode(
+            jwe.decrypt(body, settings.JWE_SECRET),
+            settings.JWT_SECRET,
+            options={'verify_exp': False},
+            algorithm='HS256'
+        )
+    except (jwt.InvalidTokenError, TypeError):
+        raise AuthenticationFailed
+    return payload
+
+
+def get_user_status(user):
+    """
+    Get the status of a given user/account.
+
+    :param user: the user instance
+    :return: the user's status in String
+    """
+
+    try:
+        check_user(user)
```

Comment:
`get_user_status(user)` uses `api.base.authentication.drf.check_user(user)`, which is a comprehensive user status check  from https://github.com/CenterForOpenScience/osf.io/pull/6832